packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -97,9 +100,31 @@ export class OpenId4VcSiopVerifierService {
     let clientIdScheme: ClientIdScheme
     let clientId: string
 
+    let authorizationResponseUrl = joinUriParts(this.config.baseUrl, [
+      options.verifier.verifierId,
+      this.config.authorizationEndpoint.endpointPath,
+    ])
+
     if (jwtIssuer.method === 'x5c') {
-      clientId = jwtIssuer.issuer
-      clientIdScheme = jwtIssuer.clientIdScheme
+      const leafCertificate = X509Service.getLeafCertificate(agentContext, { certificateChain: jwtIssuer.x5c })
+
+      if (leafCertificate.sanDnsNames.includes(getDomainFromUrl(jwtIssuer.issuer))) {
+        clientIdScheme = 'x509_san_dns'
+      } else if (leafCertificate.sanUriNames.includes(jwtIssuer.issuer)) {
+        clientIdScheme = 'x509_san_uri'
+      } else {
+        throw new CredoError(
+          `With jwtIssuer 'method' 'x5c' the jwtIssuer's 'issuer' field must either match the match a sanDnsName (FQDN) or sanUriName in the leaf x509 chain's leaf certificate.`
+        )
+      }
+
+      if (clientIdScheme === 'x509_san_dns') {
+        clientId = getDomainFromUrl(jwtIssuer.issuer)
+        authorizationResponseUrl = jwtIssuer.issuer
```

Comment:
Do we check if the jwt.issuer matches the auth response url? I think overwriting it can lead to weird behavior, as then the endpoint isn't reachable anymore